pkg/storage/factory.go

```diff
@@ -65,6 +71,98 @@ type NamedStores struct {
 	Swift      map[string]openstack.SwiftConfig     `yaml:"swift"`
 }
 
+func (ns NamedStores) checkForDuplicates() error {
+	t := reflect.TypeOf(ns)
+	v := reflect.ValueOf(ns)
+
+	seen := make(map[string]string)
+	// iterate over the struct fields
+	for i := 0; i < t.NumField(); i++ {
+		field := t.Field(i)
+		if field.Type.Kind() != reflect.Map {
+			continue
+		}
+
+		objectType := getYAMLTagKey(field)
+		// ignore fields without a valid yaml name
+		if objectType == "" {
+			continue
+		}
+
+		for _, key := range v.FieldByIndex(field.Index).MapKeys() {
+			if key.Kind() != reflect.String {
+				break
+			}
+
+			name := key.Interface().(string)
+
+			// return error if the name matches a predefined storage type
+			switch name {
+			case config.StorageTypeAWS, config.StorageTypeAWSDynamo, config.StorageTypeS3,
+				config.StorageTypeGCP, config.StorageTypeGCPColumnKey, config.StorageTypeBigTable, config.StorageTypeBigTableHashed, config.StorageTypeGCS,
+				config.StorageTypeAzure, config.StorageTypeBOS, config.StorageTypeSwift, config.StorageTypeCassandra,
+				config.StorageTypeFileSystem, config.StorageTypeInMemory, config.StorageTypeGrpc:
+				return fmt.Errorf("named store %s should not match with the name of a predefined storage type", name)
```

Comment:
nit: i'd suggest changing these from `%s` to `%q` so that error messages will be a little more clear:

> named store aws should not match...
vs
> named store "aws" should not match...